### PR TITLE
add default values in extractor2.go

### PR DESCRIPTION
**Pull request by sudeepdino008**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -312,8 +313,8 @@ Please make sure that the --binary-file-path and --avro-codec-path matches the o
 
 #### extractor2.go
 
-similar to extractor.go, but outputs the specimen, result and block_replica in separate json files.
+similar to extractor.go, but outputs the specimen, result and block_replica in separate json files. It works with the ethereum block codec version 0.3. Whereas, extractor.go works with codec version 0.2 only. Ideally both scripts should be able to handle the different versions. 
 
 ```bash
-go run extractor2.go --binary-file-path="/Users/sudeep/repos/bsp-agent/data/block-ethereum/" --codec-path "../../codec/block-ethereum.avsc" --indent-json 0 --end-block-number 15185258 --start-block-number 15185258 --chain-id "1"
+go run extractor2.go --binary-file-path="/Users/sudeep/repos/bsp-agent/data/block-ethereum/" --codec-path "../../codec/block-ethereum.avsc" --indent-json 0 --end-block-number 15185258 --start-block-number 15185258 --chain-id "1" --output-file-path "./"
```
Comment (noslav) [inline README.md:319]: in this example both start block number and end block number are the same, that would break the logic in main.go

**Review by noslav — APPROVED** [review]
LGTM 👍🏼 . Left some comments.

**Comment by noslav** [comment]
@sudeepdino008 What do you pass to the EVM binary? the filename+".specimen.json",or filename+".result.json", and how do we know in which of the either cases it might fail?

File: README.md (hunk 1 above)
Comment (sudeepdino008) [reply README.md:319]: start and end block number are both inclusive, so it just executes on "15185258" block number.

File: scripts/replica/extractor2.go [hunk 2]
```diff
@@ -39,15 +40,11 @@ var (
...
 
-func generateTestForOneBlock(binaryFilePathFlag string,
-	avroCodecPathFlag string,
-	indentJSONFlag int,
-	startingBlockNumber int64,
-	endingBlockNumber int64,
-	chainID string) {
-	replicaSegmentFiles := filterReplicaSegmentFiles(binaryFilePathFlag, startingBlockNumber, endingBlockNumber, chainID)
+func generateTestForOneBlock() {
+	replicaSegmentFiles := filterReplicaSegmentFiles(binaryFilePathFlag, startBlockNumberFlag, endBlockNumberFlag, chainIDFlag)
```
[the remark this answers is not in the document]
Comment (sudeepdino008) [reply scripts/replica/extractor2.go:47]: it just returns the files in the file path which come within the start/end block range and chainID. Processing is done in rest of the code in generateTestForOneBlock

[the remark this answers is not in the document]
Comment (sudeepdino008) [reply scripts/replica/extractor2.go:47]: adding a comment for the filter function

**Comment by sudeepdino008** [comment]
> @sudeepdino008 What do you pass to the EVM binary? the filename+".specimen.json",or filename+".result.json", and how do we know in which of the either cases it might fail as in which of the fields might not be populated?

replica/segment block: contains all the field which is the union of result and specimen
specimen: segment minus "output fields" like receipts and logs.
result: segment minus "input fields" like state specimen (accountRead, storageRead etc.)

note that state specimen is a very large in size, and so there's a huge difference between result (couple of 10 of KBs) and specimen (around 1MB or so).

Currently, the bsp-geth outputs replica with all fields populated, but the evm tool ignores the logs and receipts, and regenerates those.

**Pull request merged**